From ef41ce7146c26d1a45d7151fb6692ff4e3b9f796 Mon Sep 17 00:00:00 2001
From: Quintin Hill <Quintin.Hill@arqiva.com>
Date: Tue, 11 Apr 2017 15:21:02 +0100
Subject: [PATCH 1/4] externals/cmake: correct Windows build

Before the ${BLAH_VER} was used before being initialised in condor_pre_external.
---
 externals/bundles/openssl/1.0.1j/CMakeLists.txt | 4 ++--
 externals/bundles/openssl/1.0.2l/CMakeLists.txt | 4 +---
 externals/bundles/pcre/8.33/CMakeLists.txt      | 3 ++-
 externals/bundles/pcre/8.40/cmakelists.txt      | 2 +-
 4 files changed, 6 insertions(+), 7 deletions(-)

diff --git a/externals/bundles/openssl/1.0.1j/CMakeLists.txt b/externals/bundles/openssl/1.0.1j/CMakeLists.txt
index 80d739274e..3b3229a8ec 100644
--- a/externals/bundles/openssl/1.0.1j/CMakeLists.txt
+++ b/externals/bundles/openssl/1.0.1j/CMakeLists.txt
@@ -24,10 +24,10 @@ if (WITH_OPENSSL)
 
   if ( NOT OPENSSL_FOUND AND NOT PROPER AND NOT LINUX AND NOT DARWIN )
   
-    set (OPENSSL_DOWNLOAD ${OPENSSL_VER}.tar.gz)
-
     condor_pre_external( OPENSSL openssl-1.0.1j "lib;include" "include/openssl/ssl.h")
 
+    set (OPENSSL_DOWNLOAD ${OPENSSL_VER}.tar.gz)
+
     if (WINDOWS)
       set (OPENSSL_INSTALL cp *.lib ${OPENSSL_INSTALL_LOC}/lib \r\n
         cp *.dll ${OPENSSL_INSTALL_LOC}/lib \r\n
diff --git a/externals/bundles/openssl/1.0.2l/CMakeLists.txt b/externals/bundles/openssl/1.0.2l/CMakeLists.txt
index 3203d55de7..d0998478c7 100644
--- a/externals/bundles/openssl/1.0.2l/CMakeLists.txt
+++ b/externals/bundles/openssl/1.0.2l/CMakeLists.txt
@@ -24,13 +24,11 @@ if (WITH_OPENSSL)
 
   if ( NOT OPENSSL_FOUND AND NOT PROPER AND NOT LINUX AND NOT DARWIN )
 
-    set (OPENSSL_DOWNLOAD ${OPENSSL_VER}.tar.gz)
     condor_pre_external( OPENSSL openssl-1.0.2l "lib;include" "include/openssl/ssl.h")
+    set (OPENSSL_DOWNLOAD ${OPENSSL_VER}.tar.gz)
 
     if (WINDOWS)
 
-      condor_pre_external( OPENSSL openssl-1.0.2l "lib;include" "include/openssl/ssl.h")
-
       set (OPENSSL_INSTALL cp lib/*.lib ${OPENSSL_INSTALL_LOC}/lib \r\n
         cp bin/*.dll ${OPENSSL_INSTALL_LOC}/lib \r\n
         cp -r include/openssl/ ${OPENSSL_INSTALL_LOC}/include \r\n )
diff --git a/externals/bundles/pcre/8.33/CMakeLists.txt b/externals/bundles/pcre/8.33/CMakeLists.txt
index 7774e88c0f..cdc61f6448 100644
--- a/externals/bundles/pcre/8.33/CMakeLists.txt
+++ b/externals/bundles/pcre/8.33/CMakeLists.txt
@@ -18,9 +18,9 @@
 
 if ( NOT PROPER AND NOT LINUX )
 
-  set (PCRE_DOWNLOAD ${PCRE_VER}.tar.gz)
   if (WINDOWS)
     condor_pre_external( PCRE pcre-8.33 "lib;include" "include/pcre.h")
+    set (PCRE_DOWNLOAD ${PCRE_VER}.tar.gz)
 		set (PCRE_INSTALL cp pcre.lib ${PCRE_INSTALL_LOC}/lib \r\n
 						  cp pcre.dll ${PCRE_INSTALL_LOC}/lib \r\n
 						  cp pcre.h ${PCRE_INSTALL_LOC}/include )
@@ -36,6 +36,7 @@ if ( NOT PROPER AND NOT LINUX )
     set (PCRE_CONFIGURE echo "No configuration necessary")
     set (PCRE_MAKE echo "No make necessary")
   else()
+    set (PCRE_DOWNLOAD ${PCRE_VER}.tar.gz)
     # Currently not attempting to deal with newer version of PCRE on other platforms
   endif()
 
diff --git a/externals/bundles/pcre/8.40/cmakelists.txt b/externals/bundles/pcre/8.40/cmakelists.txt
index c18de9f616..d1d868af6b 100644
--- a/externals/bundles/pcre/8.40/cmakelists.txt
+++ b/externals/bundles/pcre/8.40/cmakelists.txt
@@ -18,7 +18,6 @@
 
 if ( NOT PROPER AND NOT LINUX )
 
-  set (PCRE_DOWNLOAD ${PCRE_VER}.tar.gz)
   if (WINDOWS)
     condor_pre_external( PCRE pcre-8.40 "lib;include" "include/pcre.h")
 		set (PCRE_INSTALL cp pcre.lib ${PCRE_INSTALL_LOC}/lib \r\n
@@ -34,6 +33,7 @@ if ( NOT PROPER AND NOT LINUX )
     set (PCRE_CONFIGURE echo "No configuration necessary")
     set (PCRE_MAKE echo "No make necessary")
   else()
+    set (PCRE_DOWNLOAD ${PCRE_VER}.tar.gz)
     # Currently not attempting to deal with newer version of PCRE on other platforms
   endif()
 
-- 
2.13.5

